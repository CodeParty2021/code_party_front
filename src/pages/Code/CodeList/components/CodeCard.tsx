import React from "react";
import { Link } from "react-router-dom";
import styled from "styled-components";

type Props = {
  id: string;
  codeContent: string;
  updatedAt: string;
  deleteHandler: (id: string) => void;
};

const Card = styled.div`
  background-color: gray;
  margin: 10px;
  padding: 10px;
  border-radius: 5px;
  width: 600px;
  color: white;
`;

export const CodeCard: React.FC<Props> = (props: Props) => {
  return (
    <Card>
      <pre>{props.codeContent}</pre>
      <p>UpdatedAt:{props.updatedAt} </p>
      <div>
        <Link to={"/free-coding/" + props.id + "/codes"}>編集</Link>
        <button onClick={() => props.deleteHandler(props.id)}>削除</button>
      </div>
    </Card>
  );
};
