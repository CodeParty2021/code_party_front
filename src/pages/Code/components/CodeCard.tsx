import React from "react";
import styled from "styled-components";

type Props = {
  id: string;
  codeContent: string;
  updatedAt: string;
};

const Card = styled.div`
  background-color: gray;
  margin: 10px;
  padding: 10px;
  border-radius: 5px;
  width: 600px;
  color: white;
`;

export const CodeCard: React.FC<Props> = (props: Props) => {
  console.log("hellop");

  console.log("card", props);
  return (
    <Card>
      <pre>{props.codeContent}</pre>
      <p>UpdatedAt:{props.updatedAt} </p>
    </Card>
  );
};
