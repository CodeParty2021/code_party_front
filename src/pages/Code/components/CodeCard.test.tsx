import React from "react";
import { shallow } from "enzyme";
import { CodeCard } from "./CodeCard";

describe("<CodeCard />", () => {
  it("auth snapshot test", () => {
    const wrapper = shallow(
      <CodeCard id={"123"} codeContent={"123"} updatedAt={"123"} />
    );

    expect(wrapper.getElements()).toMatchSnapshot();
  });
});
