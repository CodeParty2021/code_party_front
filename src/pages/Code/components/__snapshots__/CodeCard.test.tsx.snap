// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CodeCard /> auth snapshot test 1`] = `
Array [
  <styled.div>
    <pre>
      123
    </pre>
    <p>
      UpdatedAt:
      123
       
    </p>
  </styled.div>,
]
`;
